datalad/support/gitrepo.py

```diff
@@ -3444,6 +3468,49 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
         # TODO yield result for commit, prev helper checked hexsha pre
         # and post...
 
+    def _check_for_win_compat(self, files):
+        """Check file names for illegal characters or reserved names on Windows
+
+        In the case that a non-Windows-compatible file is detected, warn users
+        about potential interoperability issues.
+        """
+        from collections import defaultdict
```

Comment:
I guess if `'datalad.save.windows-compat-warning'` is `'none'` there is no point to go through all of this at all, right? then better just to return here if `none`

Reply:
I'm probably misunderstanding, but I thought we wouldn't go through this in that case because of the 
```
 if self.config.obtain("datalad.save.windows-compat-warning") in \
                    ('warning', 'error'):
 ``` 
 
 conditional above? Or should I add an explicit 
 
 ```
 if  self.config.obtain("datalad.save.windows-compat-warning") == "none":
    return
``` 

at the start of the method?

Reply:
oh, right -- it is done outside of this function.  I guess that is what confused me -- spread around logic.  Might be worth indeed "centralize" by
- bind `self.config.obtain("datalad.save.windows-compat-warning")` to a local variable within function (unlikely to feel effect on performance but still -- why to ask 3 times for the same thing?)
- moving this conditional inside  and `return to_add, None`
?

Reply:
thanks, I have applied your suggestions